### PR TITLE
feat: new module network/connection

### PR DESCRIPTION
**Pull request by fabmas**
## Description

New network/connection module migrated from CARML to AVM

[avm.res.network.connection](https://github.com/fabmas/bicep-registry-modules/actions/workflows/avm.res.network.connection.yml)

### CODE REVIEW

File: avm/res/network/connection/main.bicep [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  ipsecEncryption: ''
+  ipsecIntegrity: ''
+  ikeEncryption: ''
+  ikeIntegrity: ''
+  dhGroup: ''
+  pfsGroup: ''
+}
+
+@description('Optional. The weight added to routes learned from this BGP speaker.')
+param routingWeight int = -1
```
Comment (AlexanderSehr) [inline avm/res/network/connection/main.bicep:72]: Should be changed to nullable as the -1 was just a workaround

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: avm/res/network/connection/tests/e2e/defaults/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    name: '${namePrefix}${serviceShort}001'
+    virtualNetworkGateway1: {
+      id: nestedDependencies.outputs.primaryVNETGatewayResourceID
+    }
+    virtualNetworkGateway2: {
+      id: nestedDependencies.outputs.secondaryVNETGatewayResourceID
+    }
+    connectionType: 'Vnet2Vnet'
+    vpnSharedKey: password
+    tags: {
```
Comment (AlexanderSehr) [inline avm/res/network/connection/tests/e2e/defaults/main.test.bicep:71]: Remove as optional

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: avm/res/network/connection/tests/e2e/vnet2vnet/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,74 @@
+targetScope = 'subscription'
```
Comment (AlexanderSehr) [inline avm/res/network/connection/tests/e2e/vnet2vnet/main.test.bicep:1]: I wonder if this test provides any value beyond the default test that also tests VNET to VNET

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: avm/res/network/connection/tests/e2e/vnet2vnet/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+metadata name = 'Using VNet-to-VNet parameter set'
+metadata description = 'This instance deploys the module with the VNet-to-VNet set of required parameters.'
```
Comment (AlexanderSehr) [inline avm/res/network/connection/tests/e2e/vnet2vnet/main.test.bicep:4]: If you say 'required' here, I have to ask if the bgb parameter must be specified with false for VNET to VNET

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: avm/res/network/connection/tests/e2e/vnet2vnet/main.test.bicep (hunk 3 above)
Comment (fabmas) [reply avm/res/network/connection/tests/e2e/vnet2vnet/main.test.bicep:1]: I would ask you if you agree to remove the entire vnet2vnet folder

Comment (AlexanderSehr) [reply avm/res/network/connection/tests/e2e/vnet2vnet/main.test.bicep:1]: That's kinda my suggestion. So if you agree with me - go ahead @fabmas ;)

File: avm/res/network/connection/tests/e2e/vnet2vnet/main.test.bicep (hunk 4 above)
Comment (fabmas) [reply avm/res/network/connection/tests/e2e/vnet2vnet/main.test.bicep]: test folder removed

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**